### ISSUE TITLE
An expect/actual function with default arguments and a Composable argument fails to run or compile

### ISSUE THREAD

**Issue by ShreckYe** [JetBrains/compose-multiplatform#2806]
**Describe the bug**
Calling an expect/actual function with adefault arguments and a Composable argument throws a `NoSuchMethodError` during desktop (JVM) `run` and fails to build with an `AssertionError` on JS.

**Affected platforms**
Select one of the platforms below:
- Desktop
- Web (K/JS) - HTML based API
- Possbly others too but I didn't test all of them

**Versions**
- Kotlin version*: 1.8.0
- Compose Multiplatform version*: 1.3.0
- OS version(s)* (required for Desktop and iOS issues): Ubuntu 22.04.2 LTS
- OS architecture (x86 or arm64): x86-64
- JDK (for desktop issues): OpenJDK Runtime Environment (build 17.0.5+8-Ubuntu-2ubuntu122.04)

**To Reproduce**
Steps and/or the code snippet to reproduce the behavior:
1. Check out the code at <https://github.com/huanshankeji/compose-multiplatform-material/releases/tag/list-item-for-expect-actual-default-argument-issue> and see the function `itemForIssue`
2. Run the gradle tasks `:demo:run` and `:demo:jsBrowserRun`
3. See error

**Expected behavior**
The code should build and run without exceptions.

**Comment by igordmn** [comment]
Thanks!

We had [the similar issue](https://github.com/JetBrains/compose-jb/issues/758) in the past. But it seems the bug is not complitly fixed.

This is still reproduced:
```
// commonMain
expect fun itemInternal(
    key: Any? = null,
    content: @Composable () -> Unit
)

// desktopMain
actual fun itemInternal(
    key: Any?,
    content: @Composable () -> Unit
) {

}

fun main() = singleWindowApplication {
    itemInternal {

    }
}
```

If we add `@Composable` to `itemInternal`, the issue dissapears.

**Comment by eymar** [comment]
Fixed in 1.4.0 if used with kotlin 1.8.20

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2806">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
